sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_generated/v3_0/aio/operations/_text_analytics_client_operations.py

```diff
@@ -47,7 +47,9 @@ async def entities_recognition_general(
         :raises: ~azure.core.exceptions.HttpResponseError
         """
         cls = kwargs.pop('cls', None)  # type: ClsType["models.EntitiesResult"]
-        error_map = {404: ResourceNotFoundError, 409: ResourceExistsError}
+        error_map = {
+            401: ClientAuthenticationError, 404: ResourceNotFoundError, 409: ResourceExistsError
```

Comment:
I don't believe that a 401 response should map to `ClientAuthenticationError` - that is for when we fail to get a token from AAD etc. 

@chlowell , can you please confirm/deny?

Reply:
@johanste in my experience, this has been the mapping, @lmazuel and I GAd autorest python with this mapping.

401 also corresponds to `ClientAuthenticationError` in TA before this change (code [here](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers.py#L46)

Reply:
This seems reasonable to me. The service responding 401 indicates an authentication problem. I'm uncertain there's great value in raising the more specific error though. If the client got a 401, that means the credential provided a token the service rejected, e.g. because a service principal doesn't have permission to access the resource. What action would an application take to recover from that?

Reply:
`ClientAuthenticationError` was already what TA was throwing before I joined, but my thinking for this is a more descriptive error. We also already GAd with this behavior, so we also can't change it in the library

Reply:
Didn't we distinguish between "I couldn't get a token to present to the service" and "I got a token that I could present to the service, but the service still doesn't like me"? I may have gotten things mixed up here....

Reply:
Not to my knowledge. You might be thinking of the way credentials distinguish between "the authority rejected my token request" and "I don't have what I need to request a token".